### ISSUE TITLE
GA/ExperimentX89: the little bomb sound always plays instead of only playing when the player is focussed

### ISSUE THREAD

**Issue by olbat** [Immemor/RaidCore#159]
The little bomb sound always plays instead of only playing when the player is focussed.
I think that this bug was probably introduced in https://github.com/NielsH/RaidCore/commit/e84634766a458984c7d6ee9e351d46010ec968d5 .

I'm not sure but I think it's an issue related to operator's priority in:

``` lua
local sSound = pName == GetPlayerUnit():GetName() and mod:GetSetting("SoundBigBomb") and "Destruction"
```

and

``` lua
local sSound = nId == GetPlayerUnit():GetId() and mod:GetSetting("SoundLittleBomb") and "RunAway"
```

(line 135 and 142)

**Comment by Immemor** [comment]
Hum, good question.

I would like to say "yes it is", but I have look on the LUA documentation ( http://www.lua.org/manual/5.1/manual.html#2.5.6 )

> Operator precedence in Lua follows the table below, from lower to higher priority:
> 
> ```
>  or
>  and
>  <     >     <=    >=    ~=    ==
>  ..
>  +     -
>  *     /     %
>  not   #     - (unary)
>  ^
> ```
> 
> As usual, you can use parentheses to change the precedences of an expression. The concatenation ('..') and exponentiation ('^') operators are right associative. All other binary operators > are left associative.

The '==' have a **bigger priority** than the 'and'

With a test scenario, we should have:
1. "a name" == "another name" => **false**
2. **false** and true => **false**
3. **false** and "Destruction" => false

In same times, when I try on "demo lua" website, obviously that works correctly.

**Comment by Immemor** [comment]
When I read again this commit (https://github.com/NielsH/RaidCore/commit/e84634766a458984c7d6ee9e351d46010ec968d5), I have the feeling, that the sound is not "Destruction" or "RunAway".

But more the count down of the little bomb timer. This last is based only on the settings.

**Comment by olbat** [comment]
Erm you're right this writing seems to be ok it must come from something else ...
Sadly it's the only big difference I found in the last commits that can cause such a bug :s

**Comment by olbat** [comment]
You're right, I think it's the counter sound that was played at each bomb and regarding to the code it's perfectly normal. Is there any chance that this sound was not played before the last modifications ?

 BTW, I'm pretty sure that something is different with the sounds on this boss, I'll be careful during next week's down and keep you aware of it :)

**Comment by Immemor** [comment]
you added this count down.

**Comment by olbat** [comment]
Oh, sorry if I did by mistake :s I have issues to see where I did it, can you tell me ? (maybe at line 144 of https://github.com/NielsH/RaidCore/commit/e84634766a458984c7d6ee9e351d46010ec968d5 ?)

**Comment by Immemor** [comment]
It's the line 144, you just need to add more condition.

And maybe factorize this condition with the sound.

**Comment by Immemor** [comment]
I completely forgotten this issue.

It's fixed in the commit: 3320d20933bf7f8cbc40812486d4a24768f46fb2
And so delivered from the v4.6 version

**Comment by olbat** [comment]
Okay thank you, and sorry for the mistake I thought that AddMsg was the only function to play sounds.
